### ISSUE TITLE
Start-test event is not getting fired with Ignored parametrized test

### ISSUE THREAD

**Issue by ashishdawale20** [nunit/nunit#3352]
I am facing the issue with ignored parametrized test. start-test event is not getting fired. Below is the scenario. 
When I run below test with console runner the sequence of events are:
1. start-run ---> start run
2. start-suite  ---> start suite
3. test-case  --> end test
4. test-suite --> end suite
5. test-run ---> END run
 
```
 [Test, Timeout(10), Author("ABC")]
 [Ignore("ignored")]
 [TestCase("ABC")]
 [TestCase("XYZ")]
 public void Test(string param)
 {
 }
```
When i change the test to below the events are correct. I have start-test event with below test having ignored with TestCase attribute clubed  

1. start-run ---> start run
2. start-suite  ---> start suite
3. start-test  --> start test
4. test-case  --> end test
5. test-suite --> end suite
6. test-run ---> END run
```
[Test, Timeout(10), Author("ABC")]
[TestCase("ABC", Ignore = "ignored")]
[TestCase("XYZ", Ignore = "ignored")]
public void Test(string param
{
}
```

**Comment by nvborisenko** [comment]
start-case event is not fired if test failed in [OneTimeSetup] method.

**Comment by CharliePoole** [comment]
@nborisenko Correct. Because no cases run in that situation.

I think this is a bit different from the situation that @ashishdawale20 is reporting.

**Comment by nvborisenko** [comment]
Generally this is the same. I think @ashishdawale20 is developing extension with ITestEventListener, and he meet situation to finish some test, but this test even didn't started. This situation can be easily avoided, just trigger start-handler yourself if unknown test is going to be finished.

I see 2 options: or resolve missing events on framework side to guarantee correct events flow; or leave comment in docs about this strange behavior.

**Comment by CharliePoole** [comment]
I understand why it seems strange and I think (as I said above) something should be done to make it consistent. However, it is equally strange that a test that never started has no test-start!

I agree that some documentation is needed. Would you suggest this should be in the docs for `ITestEventListener`? We could warn that you may get a `test-case` without a `test-start` depending on the particular framework you are using. We could even say "For example, with nunit 3, ... etc."

**Comment by CharliePoole** [comment]
I should add, btw, that this is exactly why the events are called `test-case` and `test-suite` rather than `end-test` and `end-suite`! But obviously that is not enough for documentation.

**Comment by ashishdawale20** [comment]
Yes, I faced this issue while developing extension with ITestEventListener. As a workaround, I have added ignore for each test case which solved my problem. But as @CharliePoole  is saying, it should be consistent in both the cases I stated above or documented.

**Comment by ashishdawale20** [comment]
@CharliePoole  - The effect of this bug is that the startdate and enddate for the problematic scenario are returned as "0001-01-01 00:00:00".

 ```
          [Test, Ignore("Ignore")]
              [TestCase(1)]
              [TestCase(2)]
              [TestCase(3)]
              public void SkippedTest(int a)
              {
                     Assert.Pass();
              }

Returns below XML:

<test-case id="0-7677" name="SkippedTest(3)" fullname="SkippedTest(3)" methodname="SkippedTest" classname="TestClass" runstate="Runnable" seed="173988073" result="Skipped" label="Ignored" site="Parent" start-time="0001-01-01 00:00:00Z" end-time="0001-01-01 00:00:00Z" duration="0.000000" asserts="0">

```
And when it is written like below. 

```
            [TestCase(1, Ignore = "ignored")]
              [TestCase(2, Ignore = "ignored")]
              [TestCase(3, Ignore = "ignored")]
              public void SkippedTest(int a)
              {
                     Assert.Pass();
              }


Returns below  XML  with the correct dates:

<test-case id="0-88278" name="SkippedTest(3)" fullname="SkippedTest(3)" methodname="SkippedTest" classname="TestClass" runstate="Ignored" seed="1057688393" result="Skipped" label="Ignored" start-time="2020-06-07 04:42:33Z" end-time="2020-06-07 04:42:33Z" duration="0.004777" asserts="0">

```

**Comment by CharliePoole** [comment]
@nunit/framework-team Can someone review this and either accept or reject it as an issue?

**Comment by mikkelbu** [comment]
Disclaimer, I have not worked on this area of the the code and have never used the events. That being said I think the two cases should be handled consistently. Regarding the "empty" start-time and end-time, then it sounds like a duplicate of nunit/nunit#2339 - or at least a very similar issue.

**Comment by ashishdawale20** [comment]
Hello @nvborisenko, I wanted to fix this from long time, I spent some time today debugging to understand where exactly these events get added into the queue but could not locate the code.

It will be helpful if you can help me to point out the code which does that and which is skipping start-test event for "ParameterizedMethod".

I noticed that debugger comes here:https://github.com/nunit/nunit/blob/3630a95685e1cafa8e6f89a577897ecbfe6d8f48/src/NUnitFramework/framework/Internal/TestProgressReporter.cs#L47

But never goes https://github.com/nunit/nunit/blob/3630a95685e1cafa8e6f89a577897ecbfe6d8f48/src/NUnitFramework/framework/Internal/TestProgressReporter.cs#L83 for my below test case:

 [Test, Timeout(10), Author("ABC")]
 [Ignore("ignored")]
 [TestCase("ABC")]
 [TestCase("XYZ")]
 public void Test(string param)
 {
 }

**Comment by nvborisenko** [comment]
I think you can start journey here: https://github.com/nunit/nunit/blob/3630a95685e1cafa8e6f89a577897ecbfe6d8f48/src/NUnitFramework/tests/Internal/TestProgressReporterTests.cs#L31C9-L31C9

**Comment by CharliePoole** [comment]
Note that some tests can actually start (be run) and then use Assert.Ignore to be ignored. It seems as Assert.Ignore and IgnoreAttribute ought to be reported in the same way.

**Comment by ashishdawale20** [comment]
> I think you can start journey here: https://github.com/nunit/nunit/blob/3630a95685e1cafa8e6f89a577897ecbfe6d8f48/src/NUnitFramework/tests/Internal/TestProgressReporterTests.cs#L31C9-L31C9

Thank you! Let me try
